keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -141,6 +145,13 @@ public static boolean isScalar(int[] shape) {
         return dims;
     }
 
+    public static int[] moveAxis(int from, int to, int[] shape) {
```

Comment:
I know you usually use this to move axis by passing zero as one of the parameters but it could really be used for all sorts. A better name is probably: `moveDimensions`. Or changing this to only take a `from` parameter and assume `to` is always `0`

Reply:
Renamed to `slideDimension`